framework/src/modules/token/cc_commands/cc_transfer.ts

```diff
@@ -49,104 +46,154 @@ export class CCTransferCommand extends BaseCCCommand {
 		return CROSS_CHAIN_COMMAND_NAME_TRANSFER;
 	}
 
-	public addDependencies(interoperabilityMethod: InteroperabilityMethod) {
-		this._interopMethod = interoperabilityMethod;
+	public init(args: { ownChainID: Buffer }): void {
+		this._ownChainID = args.ownChainID;
 	}
 
-	public init(args: { minBalances: MinBalance[]; supportedTokenIDs: Buffer[] }): void {
-		this._supportedTokenIDs = args.supportedTokenIDs;
-		// this._minBalances = args.minBalances;
-	}
-
-	public async execute(ctx: CCCommandExecuteContext): Promise<void> {
+	public async verify(ctx: CCCommandExecuteContext): Promise<void> {
 		const { ccm } = ctx;
 		const methodContext = ctx.getMethodContext();
-		const { id: ownChainID } = await this._interopMethod.getOwnChainAccount(methodContext);
+		const ownChainID = this._ownChainID;
 		let params: CCTransferMessageParams;
-		let tokenChainID;
-		// let tokenLocalID;
+		const mainChainID = this._tokenMethod.getMainchainTokenID();
+		let tokenChainID: Buffer;
+		const { sendingChainID } = ccm;
+
 		try {
 			params = codec.decode<CCTransferMessageParams>(crossChainTransferMessageParams, ccm.params);
 			validator.validate(crossChainTransferMessageParams, params);
+			const { tokenID, amount } = params;
+			[tokenChainID] = splitTokenID(tokenID);
+
+			if (
+				!tokenChainID.equals(ownChainID) &&
+				!tokenChainID.equals(sendingChainID) &&
+				!tokenChainID.equals(mainChainID)
+			) {
+				throw new Error(
+					'Token must be native to either the sending or the receiving chain or the mainchain.',
+				);
+			}
 
-			[tokenChainID] = splitTokenID(params.tokenID);
 			if (tokenChainID.equals(ownChainID)) {
+				// TODO: recheck if this is correct
+				// if escrowStore(sendingChainID, tokenID) does not exist
+				// or escrowAmount(sendingChainID, tokenID) < amount:
+				// raise Exception("Insufficient balance in escrow account.")
```

Comment:
the behavior will be correct, but you can also do `this.stores.get(EscrowStore).getOrDefault`, to make it more clear